src/jshint.js

```diff
@@ -2743,6 +2743,9 @@ var JSHINT = (function() {
           break;
         }
       } else {
+        if (state.option.trailingcomma && state.inES5()) {
```

Comment:
Do you think the ES5 check should be removed? (Continuing from here: https://github.com/jshint/jshint/pull/2100#discussion_r89679455.)

From a language purist perspective, it should be. Although for practical reasons it is probably still useful for any sorry soul who still has to support old IE. I don't see an option specifically for detecting old IE bugs, otherwise I'd suggest that instead.